app/controllers/mixins/actions/vm_actions/resize.rb

```diff
@@ -3,16 +3,26 @@ module Actions
     module VmActions
       module Resize
         def resizevms
-          assert_privileges("instance_resize")
-          recs = find_checked_items
+          assert_privileges(params[:pressed])
```

Comment:
Definitely not!

We can not allow the user to pick which RBAC feature to check for.

This would need a whitelist at least, but .. given the changes, maybe you need a

Reply:
@himdel I'm simply copying what's done in reconfigure.rb, which was the primary model for these changes (since both resize and reconfigure use the same UI, exposed as 'reconfigure') -- also, grepping for this, this seems to be a very common pattern. If this isn't the right way to do this, what exactly should be done here? Part of what makes these forms difficult is that the same form has to be used both for the initial request (in the context of the instance/VM ui) as well as by an admin on editing the request (in the context of the requests UI).

Reply:
Yes, sorry, it's a pattern we're trying very hard to get rid of.

So... please fix this, by working on this you are in a better position to know the possible privileges and values for `params[:pressed]` than anyone else looking at it later.

Something like..

```
case params[:pressed]
when "your_new_thing"
  assert_privileges("your_new_thing")
else
  assert_privileges("instance_resize")
end
```

would be enough.

Reply:
OK. That works for me.